p2p/module.go

```diff
@@ -333,13 +350,62 @@ func (m *p2pModule) handlePocketEnvelope(pocketEnvelopeBz []byte) error {
 		return fmt.Errorf("decoding network message: %w", err)
 	}
 
+	if m.isNonceAlreadyObserved(poktEnvelope.Nonce) {
+		// skip passing redundant message to application layer
+		return nil
+	}
+
+	if err := m.observeNonce(poktEnvelope.Nonce); err != nil {
+		return fmt.Errorf("pocket envelope nonce: %w", err)
+	}
+
+	// NB: Explicitly constructing a new `PocketEnvelop` literal with content
+	// rather than forwarding `poktEnvelope` to avoid blindly passing additional
+	// fields as the protobuf type changes. Additionally, strips the `Nonce` field.
 	event := messaging.PocketEnvelope{
 		Content: poktEnvelope.Content,
 	}
 	m.GetBus().PublishEventToBus(&event)
 	return nil
 }
 
+// observeNonce adds the nonce to the deduper if it has not been observed.
+func (m *p2pModule) observeNonce(nonce utils.Nonce) error {
+	// Add the nonce to the deduper
```

Comment:
Can we just do `return m.nonceDeduper.Push(nonce)`

Reply:
:thinking: We could. I would prefer just simplifying this definition:

```
// observeNonce adds the nonce to the deduper if it has not been observed.
func (m *p2pModule) observeNonce(nonce utils.Nonce) error {
	// Add the nonce to the deduper
	return m.nonceDeduper.Push(nonce)
}
```

I think having `#observeNonce()` and `#isNonceObserved()` encapsulate the details of interacting with the nonce deduper follows the [single-responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle). Even though the API of the nonce deduper must be known to the p2p module, it's not necessary to expose/interact with it more than necessary; giving every other piece of code which references the deduper an additional "reason to change".

Reply:
Perhaps a even better alternative would be to change `NonceDeduper` to no longer be an alias and them move these two methods to it instead. :thinking:

Reply:
Completely on the same page regarding the wrapper :)